### PR TITLE
ARROW-3267: [Python] Create empty table from schema

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

**Comment by kszucs** [comment]
I wonder whether classmethods would provide better usability, have you considered e.g.:
 - `Schema.as_table()`
 - `Table.from_schema(schema)`
 - `Table.empty(schema)`
?

**Comment by xhochy** [comment]
I was tempted to create `Schema.empty_table()`

**Comment by kszucs** [comment]
That feels better :)

**Comment by xhochy** [comment]
@kszucs updated the name.

File: python/pyarrow/tests/test_types.py [hunk 1]
```diff
@@ -461,3 +461,13 @@ def test_field_add_remove_metadata():
...
+
+
+def test_empty_table():
+    schema = pa.schema([
+        pa.field("oneField", pa.int64())
+    ])
+    table = schema.empty_table()
+    assert isinstance(table, pa.Table)
+    assert table.num_rows == 0
+    assert schema == schema
```
Comment (kszucs) [inline python/pyarrow/tests/test_types.py]: `table.schema`

Comment (xhochy) [reply python/pyarrow/tests/test_types.py]: 🤦‍♂️

Comment (kszucs) [reply python/pyarrow/tests/test_types.py]: :)

**Review by kszucs — APPROVED** [review]

**Comment by kszucs** [comment]
Merging on green

**Pull request closed without merging**